packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -769,7 +770,10 @@ export class CdkToolkit {
       defaultBehavior: DefaultSelection.OnlySingle,
     });
 
-    // No validation
+    const notExistPatterns = selector.patterns.filter(p => !stacks.stackArtifacts.find(s => minimatch(s.hierarchicalId, p)));
+    if (notExistPatterns.length > 0) {
+      throw new Error(`Stacks not exist: ${notExistPatterns.join(', ')}`);
```

Comment:
Can you please change the error message to be something more informative as to what does not exist and how they might be able to fix it?

Reply:
@SankyRed 

For example, if you specify stacks `stack-A`, `stack-X`, and `stack-Y` in `cdk destroy`, and only `stack-A` exists in your account.

In this code, the error message is to be following, `Stacks not exist: stack-X, stack-Y`. So you can know which of stacks do not exist.

`how they might be able to fix it?`

It is to difficult to know this. If you enter non-existent stacks, CDK does not have the way to fix it unless it gets all the stacks in the AWS account and does a fuzzy search on all of them. Even if we did that, you probably wouldn't get the expected and accurate output.

Reply:
A tad more context here might be helpful, if we can provide it. Maybe something like `Cannot run cdk destroy on stack(s) ${stacks.join(', ')}. ${notExistPatterns.join(', ')} are not deployed in account: ${account}, region: ${region}.`

Reply:
@TheRealAmazonKendra 

> A tad more context here might be helpful, if we can provide it. Maybe something like Cannot run cdk destroy on stack(s) ${stacks.join(', ')}. ${notExistPatterns.join(', ')} are not deployed in account: ${account}, region: ${region}.

I think the format is difficult.

This `notExistPattern` handling and `cdk destroy` check the non-existence stacks by looking not the AWS account but stack instance definitions in the cdk app (that is cloud-assembly).

My understanding is that `cdk destroy` does not run the command by specifying an account or region, but looks at the stack list ([`CloudFormationStackArtifact[]`](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/cx-api/lib/artifacts/cloudformation-artifact.ts)) in cloud-assembly and if each stack has a region and account tied to it If it is tied to a region and account, it is used; otherwise, the default region is used.
So we know that the stack entered does not exist in the CDK app in the first place, but we do not know which account/region it does not exist in.

Then, how about the following message?:

```
Cannot run cdk destroy on stack(s) ${stacks.join(', ')}. ${notExistPatterns.join(', ')} not exist.
```